### PR TITLE
[ML] Add REST spec for the update data frame analytics endpoint

### LINKED ISSUE #59148: `_ml/data_frame/analytics/{id}/_update` is not present in rest-api-spec
Hello team!
While the API `_ml/data_frame/analytics/{id}/_update` is documented in the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/master/update-dfanalytics.html), it appears not to be present in the [rest-api-spec](https://github.com/elastic/elasticsearch/tree/master/x-pack/plugin/src/test/resources/rest-api-spec/api) (both `master` and `7.x`), is there a reason for this?
Thanks!

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**
Closes #59148

### CODE REVIEW

File: x-pack/plugin/src/test/resources/rest-api-spec/test/ml/data_frame_analytics_crud.yml [hunk 1]
```diff
@@ -2100,3 +2100,53 @@ setup:
...
+  - match: { description: "after update" }
+  - match: { model_memory_limit: "30mb" }
+  - match: { allow_lazy_start: true }
+
+---
+"Test update given missing analytics":
+
+  - do:
+      catch: missing
+      ml.update_data_frame_analytics:
```
Comment (benwtrent) [inline x-pack/plugin/src/test/resources/rest-api-spec/test/ml/data_frame_analytics_crud.yml:2147]: I don't see a `update_data_frame_analytics.json` definition file.

Comment (przemekwitek) [reply x-pack/plugin/src/test/resources/rest-api-spec/test/ml/data_frame_analytics_crud.yml:2147]: Me neither.
Apart from that, the test looks good!

Comment (dimitris-athanasiou) [reply x-pack/plugin/src/test/resources/rest-api-spec/test/ml/data_frame_analytics_crud.yml:2147]: Yeah, I never added it :-(

**Review by przemekwitek — APPROVED** [review]
LGTM

File: x-pack/plugin/src/test/resources/rest-api-spec/api/ml.update_data_frame_analytics.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
+  "ml.update_data_frame_analytics":{
+    "documentation":{
+      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/update-df-analytics.html",
```
Comment (delvedor) [inline x-pack/plugin/src/test/resources/rest-api-spec/api/ml.update_data_frame_analytics.json:4]: It looks like the docs url already publish on elastic.co is different:
```diff
- https://www.elastic.co/guide/en/elasticsearch/reference/current/update-df-analytics.html
+ https://www.elastic.co/guide/en/elasticsearch/reference/current/update-dfanalytics.html
```

Comment (dimitris-athanasiou) [reply x-pack/plugin/src/test/resources/rest-api-spec/api/ml.update_data_frame_analytics.json:4]: Ah, well spotted. I'll fix it.

**Pull request merged**